setup.sh

```diff
@@ -1,7 +1,24 @@
+#!/bin/bash
 go install google.golang.org/protobuf/cmd/protoc-gen-go@latest
-mkdir -p ~/opt && mkdir -p ~/opt/semantifly && cd ~/opt/semantifly
+mkdir -p ~/opt/semantifly && cd ~/opt/semantifly
+
+# Download and install protoc
 wget https://github.com/protocolbuffers/protobuf/releases/download/v27.3/protoc-27.3-linux-x86_64.zip
 unzip protoc-27.3-linux-x86_64.zip
-export PATH=$PATH:~/opt/semantifly/bin
-echo "Protoc installed to ~/opt/semantifly/bin: fix pending to add to PATH"
-echo "For now, invoke as eg '~/opt/semantifly/bin/protoc -I=src/proto --go_out=src/proto src/proto/index.proto'"
+rm protoc-27.3-linux-x86_64.zip
+
+cd -
+cd src
+go get -d ./...
+
+# Build binary, move to PATH
+go build -o ~/opt/semantifly/semantifly .
+echo 'export PATH="$HOME/opt/semantifly:$PATH"' >> ~/.bashrc
+
+# Update PATH
+. ~/.bashrc
+
+cd -
+find src/proto -name "*.proto" -exec ~/opt/semantifly/bin/protoc -I=src/proto --go_out=src/proto {} \;
```

Comment:
This needs to be performed before the go build command since we want to link against the generated pb.go files to build semantifly

Reply:
fixed